azerothcore/azerothcore-wotlk
# fix(Core/Spells): Improvements to Far Sight spell:

Far Sight should not interrupt while casting another spell.
Corrected setting Far Sight as an active object.
Fixed grid activation range for active dynamic objects.
When Far Sight is over, the camera should be reset to player.
Enable swapping camera between Far Sight and Sentry Totem.
Fixes #6368



## Issues Addressed:

- Closes #6368
- Closes https://github.com/chromiecraft/chromiecraft/issues/870
- Closes https://github.com/chromiecraft/chromiecraft/issues/1019

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

Use Far Sight
Use Far Sight again while the current one is still being channeled
Notice the camera
Let the second cast run out
Notice the camera
Cast again and interrupt to reset the camera

Put down a sentry totem
Cast far sight
See that the camera does move to far sight spot.
Cancel far sight
Camera is again on sentry totem


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
